File: java/EclipseWorkSpace/NEATGeneticAlgorithm/src/neat/Button.java
package neat;

public class Button {

	// Config variables
	private Float2 pos;
	private Float2 size;
	int buttonType;
	
	
	// Default constructor
	public Button(float px, float py, float sx, float sy, int buttonType_) {
		pos = new Float2(px, py);
		size = new Float2(sx, sy);
		buttonType = buttonType_;
	}

	
	// Update the button
	public void callUpdate() {
		show();
	}
	
	
	// Show the button
	public void show() {
		Main.p.strokeWeight(2);
		Main.p.stroke(180);
		Main.p.fill(mouseOntop() ? 240 : 220);
		Main.p.rect(
			pos.x, 
			pos.y,
			size.x,
			size.y
		);
	}
	
	
	// Called whe mouse pressed
	public void mousePressed() {
		if (mouseOntop()) {
			press();
		}
	}
	
	
	// Check if the PApplet mouse is ontop
	public boolean mouseOntop() {
		return Main.p.mouseX > pos.x
			&& Main.p.mouseX < pos.x + size.x
			&& Main.p.mouseY > pos.y
			&& Main.p.mouseY < pos.y + size.y;
	}
	
	
	// Button has been pressed
	public void press() {
		switch(buttonType) {
			case (0): Main.genAl.createPopulation(); break;
		
			case (1): Main.genAl.toUpdate = !Main.genAl.toUpdate; break;
				
			case (2): Main.genAl.toFinish = !Main.genAl.toFinish; break;

			case (3): Main.genAl.toLoop = !Main.genAl.toLoop; break;

			case (4): Main.genAl.updateGeneration(); break;

			case (5): Main.genAl.finishGeneration(); break;
			
			case (6): Main.genAl.loopGeneration(); break;

			case (7): Main.genAl.updatesPerFrame--; break;
			
			case (8): Main.genAl.updatesPerFrame++; break;
		}
	}
}

File: Multi/machine_learning/5 - JS - Backprop (FNN)/Seperated1/button.js

class Button {

  // Declare static variables
  static OUTLINE_COLOR = "#1d1d1f";
  static BACKGROUND_COLOR = "#777785";
  static HOVER_COLOR = "#848491";


  constructor(px, py, sx, sy, clicked, held) {
    // Initialize variables
    this.pos = { x: px, y: py };
    this.size = { x: sx, y: sy };
    this.clicked = clicked || (() => {});
    this.held = held || (() => {});
  }


  update(view, interactable) {
    // Check whether being clicked / held on
    if (interactable && this.mouseOntop(view)) {
      if (input.mouse.clicked[LEFT]) this.clicked();
      if (input.mouse.held[LEFT]) this.held();
    }
  }


  show(view) {
    // Draw button as box
    view.getOutput().strokeWeight(1);
    view.getOutput().stroke(Button.OUTLINE_COLOR);
    if (this.mouseOntop(view)) view.getOutput().fill(Button.HOVER_COLOR);
    else view.getOutput().fill(Button.BACKGROUND_COLOR);
    view.getOutput().rect(this.pos.x, this.pos.y, this.size.x, this.size.y);
  }


  mouseOntop(view) {
    // Check whether mouse is ontop
    return (
      view.getMouseX() > (this.pos.x)
      && view.getMouseX() <= (this.pos.x + this.size.x)
      && view.getMouseY() > (this.pos.y)
      && view.getMouseY() <= (this.pos.y + this.size.y)
    );
  }
}